runtime/gc_ctrl.c

```diff
@@ -572,21 +580,25 @@ CAMLprim value caml_gc_full_major(value v)
   // call finalisers
   caml_process_pending_actions();
   CAML_INSTR_TIME (tmr, "explicit/gc_full_major");
+  caml_ev_end(EV_EXPLICIT_GC_FULL_MAJOR);
```

Comment:
Idem, and here you can use a goto to avoid duplicating clean-up code.

Reply:
Thank you! Does it look right this way?

Reply:
Perfect.